examples/hetero/han_imdb.py

```diff
@@ -0,0 +1,89 @@
+from typing import Union, Dict, List
+import os.path as osp
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+from torch_geometric.datasets import IMDB
+from torch_geometric.nn import HANConv
+import torch_geometric.transforms as T
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '../../data/IMDB')
+metapaths = [[('movie', 'actor'), ('actor', 'movie')],
+             [('movie', 'director'), ('director', 'movie')]]
+transform = T.AddMetaPaths(metapaths=metapaths, drop_orig_edges=True)
+dataset = IMDB(path, transform=transform)
+data = dataset[0]
+del data['director'], data['actor']
```

Comment:
We think we should add this functionality to `AddMetaPaths` directly. If `drop_orig_edges=True`, there is really no need to maintain node types that are no longer connected by any edge type. WDYT?

Reply:
Yeah I think that's a good idea.
So if `drop_orig_edges` is True and there are node types types that are not connected by any edge type we just drop them inside the transform. 
I'll update the implementation and docs of AddMetaPaths this way then.

Reply:
Super, thank you! Let me know once this is ready to be merged :)

Reply:
I implemented the dropping of nodes in a slightly different.
Add a new argument `drop_unconnected_nodes`, which if set to true remove node types not connected by any edges type.